src/NosCore.GameObject/Providers/ItemProvider/Item/WearableInstance.cs

```diff
@@ -146,5 +149,111 @@ public void SetRarityPoint()
                     break;
             }
         }
+
```

Comment:
This single method is more than 100 lines long, maybe splitting it into distinct smaller methods would be good, such as "HandleSuccessSum" and "HandleFailedSum", just an idea though, would make the code much clearer